Replace special characters with unicode

some tests do not compile in circleci because of `unmappable character (0xBC) for encoding US-ASCII`

```
> Task :inspectit-ocelot-core:compileTestJava
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:43: error: unmappable character (0xC3) for encoding US-ASCII
                    enc("my_val??") + " =  " + enc("stra??e=15") + ";someprop=42");
                               ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:43: error: unmappable character (0xBC) for encoding US-ASCII
                    enc("my_val??") + " =  " + enc("stra??e=15") + ";someprop=42");
                                ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:43: error: unmappable character (0xC3) for encoding US-ASCII
                    enc("my_val??") + " =  " + enc("stra??e=15") + ";someprop=42");
                                                        ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:43: error: unmappable character (0x9F) for encoding US-ASCII
                    enc("my_val??") + " =  " + enc("stra??e=15") + ";someprop=42");
                                                         ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:47: error: unmappable character (0xC3) for encoding US-ASCII
            verify(inspectitContext).setData(eq("my_val??"), eq("stra??e=15"));
                                                       ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:47: error: unmappable character (0xBC) for encoding US-ASCII
            verify(inspectitContext).setData(eq("my_val??"), eq("stra??e=15"));
                                                        ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:47: error: unmappable character (0xC3) for encoding US-ASCII
            verify(inspectitContext).setData(eq("my_val??"), eq("stra??e=15"));
                                                                     ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:47: error: unmappable character (0x9F) for encoding US-ASCII
            verify(inspectitContext).setData(eq("my_val??"), eq("stra??e=15"));
                                                                      ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:106: error: unmappable character (0xC3) for encoding US-ASCII
            Map<String, Object> data = ImmutableMap.of("my_val??", "stra??e=15");
                                                              ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:106: error: unmappable character (0xBC) for encoding US-ASCII
            Map<String, Object> data = ImmutableMap.of("my_val??", "stra??e=15");
                                                               ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:106: error: unmappable character (0xC3) for encoding US-ASCII
            Map<String, Object> data = ImmutableMap.of("my_val??", "stra??e=15");
                                                                        ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:106: error: unmappable character (0x9F) for encoding US-ASCII
            Map<String, Object> data = ImmutableMap.of("my_val??", "stra??e=15");
                                                                         ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:112: error: unmappable character (0xC3) for encoding US-ASCII
                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_val??") + "=" + enc("stra??e=15"));
                                                                          ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:112: error: unmappable character (0xBC) for encoding US-ASCII
                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_val??") + "=" + enc("stra??e=15"));
                                                                           ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:112: error: unmappable character (0xC3) for encoding US-ASCII
                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_val??") + "=" + enc("stra??e=15"));
                                                                                                ^
/root/inspectit/repo/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java:112: error: unmappable character (0x9F) for encoding US-ASCII
                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_val??") + "=" + enc("stra??e=15"));
                                                                                                 ^
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

--- a/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java
+++ b/inspectit-ocelot-core/src/test/java/rocks/inspectit/ocelot/core/instrumentation/context/ContextPropagationUtilTest.java
@@ -40,11 +40,11 @@ public class ContextPropagationUtilTest {
         @Test
         public void testSingleString() {
             Map<String, String> headers = ImmutableMap.of(CORRELATION_CONTEXT_HEADER,
-                    enc("my_valü") + " =  " + enc("straße=15") + ";someprop=42");
+                    enc("my_val\u00FC") + " =  " + enc("stra\u00DFe=15") + ";someprop=42");
 
             ContextPropagationUtil.readPropagatedDataFromHeaderMap(headers, inspectitContext);
 
-            verify(inspectitContext).setData(eq("my_valü"), eq("straße=15"));
+            verify(inspectitContext).setData(eq("my_val\u00FC"), eq("stra\u00DFe=15"));
             verifyNoMoreInteractions(inspectitContext);
         }
 
@@ -103,13 +103,13 @@ public class ContextPropagationUtilTest {
         @Test
         public void testSingleString() {
 
-            Map<String, Object> data = ImmutableMap.of("my_valü", "straße=15");
+            Map<String, Object> data = ImmutableMap.of("my_val\u00FC", "stra\u00DFe=15");
 
             Map<String, String> result = ContextPropagationUtil.buildPropagationHeaderMap(data.entrySet().stream(), null);
 
             assertThat(result)
                     .hasSize(1)
-                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_valü") + "=" + enc("straße=15"));
+                    .containsEntry(CORRELATION_CONTEXT_HEADER, enc("my_val\u00FC") + "=" + enc("stra\u00DFe=15"));
         }
 
         @Test
@@ -164,6 +164,7 @@ public class ContextPropagationUtilTest {
 
             assertThat(result).isEqualTo("[]");
         }
+
         @Test
         public void emptyMap() {
             Map<String, String> data = Collections.emptyMap();

